packages/editor/src/components/rich-text/index.js

```diff
@@ -426,13 +426,14 @@ export class RichText extends Component {
 
 		const { start, end, formats } = this.createRecord();
 
-		if ( formats[ start ] ) {
-			this.props.onEnterFormattedText();
-		} else {
-			this.props.onExitFormattedText();
-		}
-
 		if ( start !== this.state.start || end !== this.state.end ) {
+			const isCaretWithinFormattedText = this.props.isCaretWithinFormattedText;
```

Comment:
Why not destruct?

Reply:
When there's only one property I don't really have a preference either way.

Reply:
I agree when it doesn't get assigned, but in this case it duplicates `isCaretWithinFormattedText`, so it does make it a bit less verbose.